Indexing/CodeTree.cpp

```diff
@@ -35,6 +35,9 @@
 namespace Indexing
 {
 
+#define GET_CONTAINING_OBJECT(ContainingClass,MemberField,object) \
+  reinterpret_cast<ContainingClass*>(reinterpret_cast<size_t>(object)-offsetof(ContainingClass,MemberField))
```

Comment:
This looks dubious to me as I don't think `size_t` is guaranteed to be big enough. What I think you want to do it cast to `char *` and then do the arithmetic on that.

Reply:
You're right, this is also implemented on [Linux](https://www.linuxjournal.com/files/linuxjournal.com/linuxjournal/articles/067/6717/6717s2.html). The main problem is with `char*` we need to `const_cast` as well, because it is called with both const and non-const arguments. Let me know which one do you think is better.

Reply:
Two macros, one for each case? Could also then be `static_cast`, I think.